consumers/src/main/java/uk/gov/moj/sdt/cmc/consumers/xml/XmlElementValueReader.java

```diff
@@ -0,0 +1,49 @@
+package uk.gov.moj.sdt.cmc.consumers.xml;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.fasterxml.jackson.dataformat.xml.XmlMapper;
+import java.io.IOException;
+import java.util.Iterator;
+import java.util.Map;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.stereotype.Component;
+
+@Slf4j
+@Component
+public class XmlElementValueReader {
+
+    public String getElementValue(String xmlContent, String xmlNodeName) {
+        JsonNode entityNode = null;
+        try {
+            XmlMapper xmlMapper = new XmlMapper();
+            JsonNode jsonNode = xmlMapper.readValue(xmlContent.getBytes(), JsonNode.class);
+            entityNode = getValuesInObject(jsonNode, xmlNodeName);
+        } catch (IOException e) {
+            log.error(e.getMessage(), e);
+        }
+        return entityNode != null ? entityNode.textValue() : "";
+    }
+
+    private JsonNode getValuesInObject(JsonNode node, String entityName) {
+        if (node == null) {
+            return null;
+        }
+        if (node.has(entityName)) {
+            return node.get(entityName);
+        }
+        if (!node.isContainerNode()) {
+            return null;
+        }
+        for (JsonNode child : node) {
+            if (child.isContainerNode()) {
+                JsonNode childResult = getValuesInObject(child, entityName);
+                if (childResult != null && !childResult.isMissingNode()) {
+                    return childResult;
+                }
+            }
```

Comment:
Does this need an else part to cover situation where child is not a container?  Should it use has() to try and get the value of the field?

Reply:
don't need that, which is covered at line number 34

Reply:
It's more what happens in the loop when the child is not a container.

The loop is only executed if the node is a container.  The loop iterates through each child and if the child is itself a container it recursively calls this method to get the entity being looked for.  If the child is not a container though it just skips to the next child.  It appears that if we have a node that's a container which has children that are not containers that it will never look at them to see if it's got the entity being looked for.  Perhaps this method should be called recursively regardless of whether the child is a container or not, i.e.

```
for (JsonNode child : node) {
    JsonNode childResult = getValuesInObject(child, entityName);
    if (childResult != null && !childResult.isMissingNode()) {
        return childResult;
    }
}
```
Might well be misunderstanding the purpose of the method though!

Reply:
changed it now